model2vec/inference/model.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import re
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import huggingface_hub
+import numpy as np
+import skops.io
+from sklearn.neural_network import MLPClassifier
+from sklearn.pipeline import Pipeline
+
+from model2vec.model import PathLike, StaticModel
+
+_DEFAULT_TRUST_PATTERN = re.compile(r"sklearn\..+")
+_DEFAULT_MODEL_FILENAME = "pipeline.skops"
+
+
+class StaticModelPipeline:
+    def __init__(self, model: StaticModel, head: Pipeline) -> None:
+        """Create a pipeline with a StaticModel encoder."""
+        self.model = model
+        self.head = head
+
+    @classmethod
+    def from_pretrained(
```

Comment:
can't we load it from the Hub? perhaps we should align the arguments a bit with the `transformers` naming given you've also adopted `from_pretrained`? 

For example using `pretrained_model_name_or_path`. https://huggingface.co/docs/transformers/v4.48.0/en/model_doc/auto#transformers.AutoTokenizer.from_pretrained

Reply:
from_pretrained loads from the hub. The arguments mimic the ones from `StaticModel` and, although they don't match transformers exactly, we're wary of introducing breaking changes.